### PR TITLE
NoNewGlobals for MapLabel

### PR DESCRIPTION
**Pull request by kinkie**
Detected by Coverity. CID 1554668: Initialization or destruction
ordering is unspecified (GLOBAL_INIT_ORDER).

### CODE REVIEW

**Review by rousskov — CHANGES_REQUESTED** [review]
When removing #1745 changes from this PR, please apply all relevant #1745 [review suggestions](https://github.com/squid-cache/squid/pull/1745#pullrequestreview-1943615284) to this PR as well.

**Review by rousskov — DISMISSED** [review]
LGTM after a minor polishing touch, thank you.

File: src/Transients.cc [hunk 1]
```diff
@@ -25,7 +25,12 @@
 #include <limits>
 
 /// shared memory segment path to use for Transients map
-static const SBuf MapLabel("transients_map");
+inline static const auto &
```
Comment (yadij) [inline src/Transients.cc:28]: I assume this was done as a side effect of @rousskov request to match/copy PR #1745 review changes.

AIUI, The reasons for using `inline static` pertain specifically to the logic being in a shared header, they become counter-productive when the logic is defined as a `static` function within a unit (`.cc` file).

```suggestion
static const auto &
```

**Review by yadij — CHANGES_REQUESTED** [review]

File: src/Transients.cc (hunk 1 above)
Comment (kinkie) [reply src/Transients.cc:28]: Fine by me; the compiler can inline if it chooses to

**Pull request closed without merging**